packages/udes-ui-validation/src/test/component.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { cleanup, render, fireEvent } from "@testing-library/react";
+import MaterialComponent from "../exampleComponent/materialUiComponent";
+
+afterEach(cleanup);
+
+describe("Validator Material UI tests", () => {
+  test("Valid state causes no change", async () => {
+    const testRenderer = render(<MaterialComponent />);
+    const numberField = testRenderer.getByTestId("materialNumberField");
+    expect(numberField["value"]).toBe("0");
+    fireEvent.change(numberField, { target: { value: "2" } });
+    expect(numberField["value"]).toBe("2");
+    fireEvent.change(numberField, { target: { value: "3" } });
+    expect(numberField["value"]).toBe("3");
+  });
+
+  test("Invalid state raises expected error", () => {
+    const testRenderer = render(<MaterialComponent />);
+    const numberField = testRenderer.getByTestId("materialNumberField");
+    expect(numberField["value"]).toBe("0");
+    fireEvent.change(numberField, { target: { value: "1" } });
+    expect(numberField["value"]).toBe("");
+    expect(testRenderer.getByText("Cannot contain a 1"));
+  });
+
+  test(
+    "Invalid first state raises expected error and marks as invalid," +
+      " preventing inspection of next rule",
+    () => {
+      const testRenderer = render(<MaterialComponent />);
+      const numberField = testRenderer.getByTestId("materialNumberField");
+      expect(numberField["value"]).toBe("0");
+      fireEvent.change(numberField, { target: { value: "99999999999" } });
+      expect(numberField["value"]).toBe("");
+      expect(testRenderer.getByText("Must be no more than 10 digits"));
+    },
+  );
+
+  test("Invalid state from custom function", () => {
+    const testRenderer = render(<MaterialComponent />);
+    const numberField = testRenderer.getByTestId("materialNumberField");
+    expect(numberField["value"]).toBe("0");
+    fireEvent.change(numberField, { target: { value: "444" } });
+    expect(numberField["value"]).toBe("");
+    expect(testRenderer.getByText("Cannot be greater than 10"));
+  });
```

Comment:
space

Reply:
which space sorry? Maybe the auto-jigging of the linter made something weird

Reply:
Ah sorry -I meant there should be a new line below to separate the tests like elsewhere